### PR TITLE
Append OwnerReference if already exists

### LINKED ISSUE #329: Always add ownerReference of parent to the resource
Creating issue to solve https://github.com/tektoncd/operator/pull/321#discussion_r649951551

/cc @nikhil-thomas

### PR DESCRIPTION
**Pull request by vinamra28**
# Changes

As of now we were adding our ownerReference if the existing
ownerReference length is equal to zero.
With this patch we will check if the existing ownerReference is same as
that of ours, if not then append our ownerReference.
See https://github.com/tektoncd/operator/pull/321#discussion_r649951551

closes #329 

Signed-off-by: vinamra28 <vinjain@redhat.com>
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes
```release-note
NONE
```

### CODE REVIEW

File: pkg/reconciler/openshift/tektonconfig/rbac.go [hunk 1]
```diff
@@ -450,6 +450,18 @@ func (r *rbac) updateClusterRoleBinding(ctx context.Context, rb *rbacv1.ClusterR
 		rb.SetOwnerReferences([]metav1.OwnerReference{r.ownerRef})
 	}
 
+	doUpdateOwnerRef := true
```
Comment (sm43) [inline pkg/reconciler/openshift/tektonconfig/rbac.go:453]: there are few other places in files where we check `len(ownerRef) ==0` and then add owner ref
while updating a resource.
should we add similar for those cases also?

Comment (sm43) [reply pkg/reconciler/openshift/tektonconfig/rbac.go:453]: @nikhil-thomas

File: pkg/reconciler/openshift/tektonconfig/rbac.go [hunk 2]
```diff
@@ -450,6 +450,18 @@ func (r *rbac) updateClusterRoleBinding(ctx context.Context, rb *rbacv1.ClusterR
...
+			doUpdateOwnerRef = false
+			break
+		}
+	}
+
+	if doUpdateOwnerRef {
+		ownerRef = append(ownerRef, r.ownerRef)
+	}
+
 	if hasSubject && (len(ownerRef) != 0) {
```
Comment (sm43) [inline pkg/reconciler/openshift/tektonconfig/rbac.go:465]: this would be now 
`if hasSubject && !doUpdateOwnerRef  {` ???

Comment (nikhil-thomas) [reply pkg/reconciler/openshift/tektonconfig/rbac.go:465]: i think `(len(ownerRef) != 0)` will catch both cases (needed update, or didn't need update).

File: pkg/reconciler/openshift/tektonconfig/rbac.go (hunk 1 above)
Comment (nikhil-thomas) [reply pkg/reconciler/openshift/tektonconfig/rbac.go:453]: @vinamra28 could you add a similar change in https://github.com/tektoncd/operator/blob/f061c934d3caf4b24ffa6f1d85f2d3658860d76c/pkg/reconciler/openshift/tektonconfig/rbac.go#L323

**Review by pradeepitm12 — COMMENTED** [review]
/lgtm

**Comment by pradeepitm12** [comment]
/hold
Would be nice to have a test for this.

**Pull request merged**